drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopup.java

```diff
@@ -162,32 +162,25 @@ public void applyChanges() {
         }
 
         //Check for unique binding
-        if ( isNew ) {
-            if ( editingCol.isBound() && !isBindingUnique( editingCol.getBinding() ) ) {
-                Window.alert( GuidedDecisionTableConstants.INSTANCE.PleaseEnterANameThatIsNotAlreadyUsedByAnotherPattern() );
+        String factBinding = editingPattern.getBoundName();
+        String fieldBinding = editingCol.getBinding();
+        if( factBinding != null && fieldBinding != null ) {
+            if ( editingCol.isBound() && ( !isBindingUnique( fieldBinding ) || factBinding.compareTo( fieldBinding ) == 0 ) ) {
+                view.warnAboutAlreadyUsedBinding();
```

Comment:
There are now 4 test methods `testBinding*` that should cover all cases. Let me know what do you think.